user_util/user_util.py

```diff
@@ -1,3 +1,33 @@
 # -*- coding: utf-8 -*-
 
 """Main module."""
+import hashlib
+
+
+RETIRED_USERNAME_FMT = 'retired_username_{}'
+RETIRED_EMAIL_FMT = 'retired_email_{}@edx.org'
+USERNAME_HASH_LENGTH = 10
+
+
+def _compute_username_hash(username, salt):
+    """
+    """
+    return hashlib.sha256(
+        salt.encode() + username.encode()
+    ).hexdigest()[:USERNAME_HASH_LENGTH]
+
+
+def get_retired_username(username, salt_list):
+    """
+    """
+    # Always use the last (most recent) salt value.
+    assert isinstance(salt_list, (list, tuple))
```

Comment:
I'd prefer not to use asserts outside of testing. While we have some in prod code currently, we shouldn't. They'll get optimized out if someone ever runs this with Python optimizations.

Reply:
True - I just worry about someone sending in a string as salt - and the "salt" ending up being the final character of the string (`salt[-1]`). Any ideas around that?

Reply:
Just change them to if's and raise a TypeError on a bad type?

Reply:
Agreed & done.